### PR TITLE
Image size validations

### PR DESCRIPTION
**Pull request by zarocknz**

### CODE REVIEW

**Comment by zarocknz** [comment]
Peter can you please take a look and let me know if any feedback? Thanks.

File: code/ZenValidatorConstraint.php [hunk 1]
```diff
@@ -1046,3 +1046,174 @@ public function getDefaultMessage()
...
+
+/**
+ * Constraint_Dimension
+ * Constrain an image field to have the specified dimension(s)
+ *
+ * @example Constraint_dimension::create('width', 100);
+ * */
+class Constraint_dimension extends ZenValidatorConstraint
+{
+    protected $type;
```
Comment (spekulatius) [inline code/ZenValidatorConstraint.php:1061]: Do you want to add some php doc blocks here?

File: code/ZenValidatorConstraint.php [hunk 2]
```diff
@@ -1046,3 +1046,174 @@ public function getDefaultMessage()
...
+    const WIDTH_HEIGHT = 'width_height';
+    const RATIO = 'ratio';
+    const MIN_WIDTH = 'min_width';
+    const MIN_HEIGHT = 'min_height';
+    const MIN_WIDTH_HEIGHT = 'min_width_height';
+    const MAX_WIDTH = 'max_width';
+    const MAX_HEIGHT = 'max_height';
+    const MAX_WIDTH_HEIGHT = 'max_width_height';
+
+    public function __construct($type,$val1,$val2=null)
```
Comment (spekulatius) [inline code/ZenValidatorConstraint.php:1093]: and here :D

File: code/ZenValidatorConstraint.php [hunk 3]
```diff
@@ -1046,3 +1046,174 @@ public function getDefaultMessage()
...
+                                return $this->ratioValidation($width, $height);
+                                break;
+                            case self::MIN_WIDTH:
+                                return $width >= $this->val1;
+                                break;
+                            case self::MIN_HEIGHT:
+                                return $height >= $this->val1;
+                                break;
+                            case self::MIN_WIDTH_HEIGHT:
+                                if (($width >= $this->val1) && ($height >= $this->val2)) {
```
Comment (spekulatius) [inline code/ZenValidatorConstraint.php]: Could be simplified to
    return (($width >= $this->val1) && ($height >= $this->val2));

File: code/ZenValidatorConstraint.php [hunk 4]
```diff
@@ -1046,3 +1046,174 @@ public function getDefaultMessage()
...
+                                }
+                                break;
+                            case self::MAX_WIDTH:
+                                return $width <= $this->val1;
+                                break;
+                            case self::MAX_HEIGHT:
+                                return $height <= $this->val1;
+                                break;
+                            case self::MAX_WIDTH_HEIGHT:
+                                if (($width <= $this->val1) && ($height <= $this->val2)) {
```
Comment (spekulatius) [inline code/ZenValidatorConstraint.php]: ditto

File: code/ZenValidatorConstraint.php [hunk 5]
```diff
@@ -1046,3 +1046,174 @@ public function getDefaultMessage()
...
+                                break;
+                            default:
+                                throw new Exception('Invalid type : ' . $this->type);
+                        }
+                    }
+                }
+            }
+        }
+
+        exit();
```
Comment (spekulatius) [inline code/ZenValidatorConstraint.php]: Is this really required?

File: code/ZenValidatorConstraint.php [hunk 6]
```diff
@@ -1046,3 +1046,174 @@ public function getDefaultMessage()
...
+                return sprintf(_t('ZenValidator.DIMRATIO', 'Image aspect ratio (shape) must be %s:%s'), $this->val1, $this->val2);
+                break;
+            case self::MIN_WIDTH:
+                return sprintf(_t('ZenValidator.DIMMINWIDTH', 'Image width must be greater than or equal to %s pixles'), $this->val1);
+                break;
+            case self::MIN_HEIGHT:
+                return sprintf(_t('ZenValidator.DIMMINHEIGHT', 'Image height must be greater than or equal to %s pixles'), $this->val1);
+                break;
+            case self::MIN_WIDTH_HEIGHT:
+                return sprintf(_t('ZenValidator.DIMMINWIDTHHEIGHT', 'Image width must be greater than or equal to %s pixles and Image height must be greater than or equal to %s pixles'), $this->val1, $this->val2);
```
Comment (spekulatius) [inline code/ZenValidatorConstraint.php]: some of the lines look pretty long, can we bring it under 100 chars? just use multiple lines...

**Comment by spekulatius** [comment]
Hello Doug, 

that looks pretty nice already. I've added a few notes here and there - nothing big...

Peter

**Comment by zarocknz** [comment]
@spekulatius Thanks. OK I think this is good go now, but let me know if anything else can be improved.

File: code/ZenValidatorConstraint.php [hunk 7]
```diff
@@ -1046,3 +1046,244 @@ public function getDefaultMessage()
...
+    protected function ratioValidation($width,$height)
+    {
+        $validationResult = false;
+
+        // Divide the width and height by the ratio, ignore the decimal points as can't have part pixels.
+        $baseWidth = floor($width / $this->val1);
+        $baseHeight = floor($height / $this->val2);
+
+        // Compare the base values, if they match then the image is in the correct aspect ratio.
+        if ($baseWidth == $baseHeight) {
```
Comment (spekulatius) [inline code/ZenValidatorConstraint.php]: this can be simplified as well:

```
 return ($baseWidth == $baseHeight);
```

**Pull request merged**